src/GovUk.Education.ExploreEducationStatistics.Admin/Services/BoundaryLevelService.cs

```diff
@@ -0,0 +1,55 @@
+#nullable enable
+using GeoJSON.Net.Feature;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Data.Model.Repository.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services;
+
+public class BoundaryLevelService(
+    IBoundaryLevelRepository boundaryLevelRepository,
+    IPersistenceHelper<StatisticsDbContext> statisticsPersistenceHelper) : IBoundaryLevelService
+{
+    public async Task<Either<ActionResult, List<BoundaryLevel>>> ListBoundaryLevels()
+    {
+        var boundaryLevels = await boundaryLevelRepository.ListBoundaryLevels();
+
+        return boundaryLevels
+            .ToList();
+    }
+
+    public async Task<Either<ActionResult, BoundaryLevel>> GetBoundaryLevel(long id)
+    {
+        var boundaryLevel = await boundaryLevelRepository.GetBoundaryLevel(id);
+
+        return boundaryLevel is not null
+            ? boundaryLevel
+            : new NotFoundResult();
+    }
+
+    public async Task<Either<ActionResult, BoundaryLevel>> CreateBoundaryLevel(
+        GeographicLevel level,
+        string label,
+        DateTime published,
+        FeatureCollection featureCollection)
+    {
+        return await boundaryLevelRepository.CreateBoundaryLevel(level, label, published, featureCollection);
+    }
+
+    public async Task<Either<ActionResult, BoundaryLevel>> UpdateBoundaryLevel(
+        long id,
+        string label)
+    {
+        return await statisticsPersistenceHelper
+            .CheckEntityExists<BoundaryLevel, long>(id)
+            .OnSuccessDo(async _ => await boundaryLevelRepository.UpdateBoundaryLevel(id, label));
+    }
```

Comment:
I know you've already mentioned the existence of the `BoundaryLevelRepository` as the reason for wanting to keep separation between the service and database logic here which proably makes sense for the `CreateBoundaryLevel` as that's a bit more involved but I think the rest of the methods would benefit from using the `StatisticsDbContext` as a dependency here rather than `IPersistenceHelper<StatisticsDbContext>`.

Recently we are attempting to avoid the use of `IPersistenceHelper` because it can be replaced with extension methods that we've added for the database context like `FirstOrNotFoundAsync` and `SingleOrNotFoundAsync`.

If you depend on the database context directly, at least the `ListBoundaryLevels`, `GetBoundaryLevel` and `UpdateBoundaryLevel` code that's split across the service and repository can all be simplified as follows:

```csharp
    public async Task<Either<ActionResult, List<BoundaryLevel>>> ListBoundaryLevels(
        CancellationToken cancellationToken = default)
    {
        return await context.BoundaryLevel.ToListAsync(cancellationToken);
    }

    public async Task<Either<ActionResult, BoundaryLevel>> GetBoundaryLevel(
        long id, CancellationToken cancellationToken = default)
    {
        return await context.BoundaryLevel.FirstOrNotFoundAsync(level => level.Id == id, cancellationToken);
    }

    public async Task<Either<ActionResult, BoundaryLevel>> UpdateBoundaryLevel(
        long id,
        string label,
        CancellationToken cancellationToken = default)
    {
        return await context.BoundaryLevel
            .FirstOrNotFoundAsync(level => level.Id == id, cancellationToken)
            .OnSuccess(async level =>
            {
                level.Label = label;
                await context.SaveChangesAsync(cancellationToken);
                return level;
            });
    }
```

Note I've also added `CancellationToken cancellationToken` to these methods as we've recently tried to start propagating `CancellationToken` down from requests.